apps/system/utils/utils_heapinfo.c

```diff
@@ -229,7 +229,8 @@ int utils_heapinfo(int argc, char **args)
 #ifdef CONFIG_APP_BINARY_SEPARATION
 		case 'b':
 			options.heap_type = HEAPINFO_HEAP_TYPE_BINARY;
-			strncpy(options.app_name, optarg, BIN_NAME_MAX);
+			strncpy(options.app_name, optarg, BIN_NAME_MAX - 1);
+			options.app_name[BIN_NAME_MAX - 1] = '\0';
```

Comment:
Do you confirm that `optarg` has `BIN_NAME_MAX - 1` size? If not, it will have garbage value in the name.

Reply:
Actually it is meaningful when `optarg` is larger than `BIN_NAME_MAX - 1 only.
But I added this to avoid calling strlen every time for performance.